File: pyg-sellergoods-interface/src/main/java/com/pyg/sellergoods/service/BrandService.java
package com.pyg.sellergoods.service;

import PageBean.PageResult;
import com.pyg.pojo.TbBrand;

import java.util.List;
import java.util.Map;

//品牌接口
public interface BrandService {
    List<TbBrand> findAll();
//    实现分页
    PageResult findPage(int page,int size);
//   添加数据
    boolean add(TbBrand brand);

    TbBrand findOne(long id);

    boolean update(TbBrand brand);

    void delete(long[] ids);

    PageResult search(TbBrand brand, int page, int size);

    List<Map> selectOptionList();
}

File: pyg-manager-web/src/main/webapp/js/Service/BrandService.js
//前端服务层
app.service('brandService', function ($http) {
    this.findAll = function () {
        return $http.get('../brand/findAll');
    }
    this.save = function (entity) {
        return $http.post('../brand/save', entity);
    }
    this.findOne = function (id) {
        return $http.get('../brand/findOne?id=' + id);
    }
    this.dele = function (ids) {
        return $http.get('../brand/delete?ids=' + ids);
    }
    this.findPage = function (page, size, searchEntity) {
        return $http.post('../brand/search?page=' + page + '&size=' + size, searchEntity);
    }
    //品牌数据查询
    this.selectOptionList=function () {
        return $http.get('../brand/selectOption');
    }
});